pkgs/servers/firebird/default.nix

```diff
@@ -1,91 +1,98 @@
-{lib, stdenv, fetchurl, libedit, automake, autoconf, libtool
-,
-  # icu = null: use icu which comes with firebird
-
-  # icu = pkgs.icu => you may have trouble sharing database files with windows
-  # users if "Collation unicode" columns are being used
-  # windows icu version is *30.dll, however neither the icu 3.0 nor the 3.6
-  # sources look close to what ships with this package.
-  # Thus I think its best to trust firebird devs and use their version
-
-  # icu version missmatch may cause such error when selecting from a table:
-  # "Collation unicode for character set utf8 is not installed"
-
-  # icu 3.0 can still be built easily by nix (by dropping the #elif case and
-  # make | make)
-  icu ? null
-
-, superServer ? false
-, port ? 3050
-, serviceName ? "gds_db"
+{ lib, stdenv, fetchFromGitHub, libedit, automake, autoconf, libtool, zlib, unzip, libtommath, libtomcrypt, icu
+  , superServer ? false
+  , port ? 3050
+  , serviceName ? "gds_db"
 }:
 
-/*
-   there are 3 ways to use firebird:
-   a) superserver
-    - one process, one thread for each connection
-   b) classic
-    - is built by default
-    - one process for each connection
-    - on linux direct io operations (?)
-   c) embedded.
-
-   manual says that you usually don't notice the difference between a and b.
-
-   I'm only interested in the embedder shared libary for now.
-   So everything isn't tested yet
-
-*/
-
-stdenv.mkDerivation rec {
-  version = "2.5.7.27050-0";
+let base = {
   pname = "firebird";
 
-  # enableParallelBuilding = false; build fails
+  meta = with lib; {
+    description = "SQL relational database management system";
+    homepage = "https://firebirdsql.org/";
+    changelog = "https://github.com/FirebirdSQL/firebird/blob/master/CHANGELOG.md";
+    license = [ "IDPL" "Interbase-1.0" ];
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ marcweber ];
+  };
 
-  # http://tracker.firebirdsql.org/browse/CORE-3246
   preConfigure = ''
     makeFlags="$makeFlags CPU=$NIX_BUILD_CORES"
   '';
 
   configureFlags =
-    [ "--with-serivec-port=${builtins.toString port}"
+    [
       "--with-service-name=${serviceName}"
       "--with-system-editline"
       "--with-fblog=/var/log/firebird"
       "--with-fbconf=/etc/firebird"
       "--with-fbsecure-db=/var/db/firebird/system"
     ]
-    ++ (lib.optional  (icu != null) "--with-system-icu")
     ++ (lib.optional superServer "--enable-superserver");
 
-  src = fetchurl {
-    url = "mirror://sourceforge/firebird/Firebird-${version}.tar.bz2";
-    sha256 = "06hp6bq5irqvm3h03s79qjgcc3jsjpq150y9aq7anklx9v4nhfqa";
-  };
-
+  configurePhase = ''
+    ./autogen.sh $configureFlags --prefix=$out
+  '';
```

Comment:
Could we use configureFlags here?

Reply:
According to firebird, yes.

- https://www.firebirdsql.org/en/building-the-code-posix/
- https://github.com/FirebirdSQL/firebird/blob/master/.github/workflows/main.yml#L38

Reply:
What do you mean?